transactions/btc.ts

```diff
@@ -66,27 +67,38 @@ export function selectUnspentOutputs(
     sumValue += pinnedOutput.value;
   }
 
-  // Sort UTXOs based on value from largest to smallest
+  // Sort UTXOs based on value from largest to smallest, deprioritising unconfirmed UTXOs
   // This will give close to the optimal spend of UTXOs to minimise fees
   unspentOutputs.sort((a, b) => {
-    const diff = b.value - a.value;
-    if (diff !== 0) {
-      return diff;
-    }
+    const aConfirmed = a.status.confirmed;
+    const bConfirmed = b.status.confirmed;
 
-    // if values are equal, we put the newer UTXO first to have a chance at CPFP
-    if (a.status.block_time && b.status.block_time) {
-      return a.status.block_time - b.status.block_time;
-    } else if (a.status.block_time) {
-      return 1;
-    } else if (b.status.block_time) {
+    // put confirmed UTXOs first
+    if (aConfirmed && !bConfirmed) {
       return -1;
+    } else if (!aConfirmed && bConfirmed) {
+      return 1;
     }
 
-    return 0;
+    return b.value - a.value;
   });
 
+  // this is slightly lower than the min vBytes of a segwit input
+  const minVBytesForInput = 70;
```

Comment:
I believe inputs sizes can vary depending on the transaction type i.e. p2wpkh or p2tr

Reply:
Yeah, that's true. This is just below the minimum of the smallest input for native segwit (which has the smallest vbytes per input). It's just a contingency check to ensure we never use extremely small UTXOs. The new logic looks at the actual size depending on the address.